ui/app/src/modules/Preview/Tools/AssetsPanel.tsx

```diff
@@ -15,24 +15,346 @@
+    dispatch(fetchAssetsPanelItems({ offset: newPage }));
+  }
+
+  function handleSearchKeyword(keyword: string) {
+    setKeyword(keyword);
+    onSearch$.next(keyword);
+  }
+
   return (
     <ToolPanel title={translations.assetsPanel}>
-      <Typography component="h2" variant="subtitle1" style={{ padding: '10px' }}>
-        Assets Panel
-      </Typography>
+      <ErrorBoundary>
+        <div className={classes.search}>
+          <SearchBar
+            onChange={handleSearchKeyword}
+            keyword={keyword}
+          />
+        </div>
+        <React.Suspense
+          fallback={
+            <LoadingState
+              title={formatMessage(translations.loading)}
+              graphicProps={{ width: 150 }}
+            />
+          }
+        >
+          {
+            dragInProgress &&
+            <div className={classes.uploadOverlay} onDrop={onDragDrop} onDragOver={onDragOver}>
+              <UploadIcon className={classes.uploadIcon}/>
+            </div>
+          }
+          <AssetsPanelUI
+            classes={classes}
+            assetsResource={resource}
+            onPageChanged={onPageChanged}
+            onDragStart={onDragStart}
+            onDragEnd={onDragEnd}
+            GUEST_BASE={GUEST_BASE}
+            onDragDrop={onDragDrop}
+            dragInProgress={dragInProgress}
+          />
+        </React.Suspense>
+      </ErrorBoundary>
     </ToolPanel>
   );
 }
+
+
+export function AssetsPanelUI(props) {
+  const {
+    classes,
+    assetsResource,
+    onPageChanged,
+    onDragStart,
+    onDragEnd,
+    GUEST_BASE
+  } = props;
+  const assets: AssetResource = assetsResource.read();
+  const { total, pageNumber, items, limit } = assets;
+  const { formatMessage } = useIntl();
+
+  return (
+    <div className={classes.assetsPanelWrapper}>
+      <TablePagination
+        className={classes.pagination}
+        classes={{ root: classes.pagination, selectRoot: 'hidden', toolbar: classes.toolbar }}
+        component="div"
+        labelRowsPerPage=""
+        count={total}
+        rowsPerPage={limit}
+        page={pageNumber}
+        backIconButtonProps={{
+          'aria-label': formatMessage(translations.previousPage),
+        }}
+        nextIconButtonProps={{
+          'aria-label': formatMessage(translations.nextPage),
+        }}
+        onChangePage={(e: React.MouseEvent<HTMLButtonElement>, page: number) => onPageChanged(e, page * limit)}
+      />
+      {
+        items.map((item: MediaItem) => {
+            return (
+              <MediaCard
+                key={item.path}
+                item={item}
+                previewAppBaseUri={GUEST_BASE}
+                hasSubheader={false}
+                avatar={DragIndicatorRounded}
+                classes={{ root: classes.card }}
+                onDragStart={onDragStart}
+                onDragEnd={onDragEnd}
+              />
+            )
```

Comment:
Check your code style settings. Set to always add semicolons. Removing the return statement and curls may fix indentation here where you don't really need the block.